ui/src/components/App/components/UserBadge/index.js

```diff
@@ -46,9 +51,18 @@ class UserBadge extends Component {
 }
 
 function mapStateToProps(state) {
-  return {
-    balance: state.balance.balance
+
+ 
+  const bal = state.balance.balance;
+  if(!this.assigned) { 
+    // If flag variable is false, keep balance as 'space' and update the flag to 'true'
+
+    state.balance.balance = "";
+    this.assigned = !this.assigned;
+  }
+  return { 
+    balance: bal   // return value of balance
   };
```

Comment:
1.needs cleanup.  should not return a value if `state.balance.balance = ""` was called
2. please check when is the earliest we can clean the local variables - like `onLoad()`